src/sentry/static/sentry/app/views/groupGrouping/mergedItem.jsx

```diff
@@ -0,0 +1,112 @@
+import React, {PropTypes} from 'react';
+import Reflux from 'reflux';
+import classNames from 'classnames';
+
+import GroupingStore from '../../stores/groupingStore';
+import GroupingActions from '../../actions/groupingActions';
+import EventOrGroupHeader from '../../components/eventOrGroupHeader';
+import EventOrGroupExtraDetails from '../../components/eventOrGroupExtraDetails';
+import SpreadLayout from '../../components/spreadLayout';
+import SplitLayout from '../../components/splitLayout';
+import Checkbox from '../../components/checkbox';
+
+const MergedItem = React.createClass({
+  propTypes: {
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+    groupId: PropTypes.string.isRequired,
+    fingerprint: PropTypes.string.isRequired,
+    disabled: PropTypes.bool,
+    event: PropTypes.shape({
+      groupID: PropTypes.string,
+      type: PropTypes.oneOf(['error', 'csp', 'default']),
+      dateCreated: PropTypes.string,
+      platform: PropTypes.string
+    })
+  },
+
+  mixins: [Reflux.listenTo(GroupingStore, 'onGroupingChange')],
+
+  getInitialState() {
+    return {
+      checked: false,
+      busy: false
+    };
+  },
+
+  onGroupingChange({unmergeState}) {
+    let {fingerprint} = this.props;
+    if (unmergeState) {
+      const stateForId = unmergeState.has(fingerprint) && unmergeState.get(fingerprint);
+      if (stateForId) {
+        Object.keys(stateForId).forEach(key => {
+          if (stateForId[key] !== this.state[key]) {
+            this.setState({
+              [key]: stateForId[key]
```

Comment:
Same comment re: avoiding deep nesting. I'd personally write this as:

```javascript
if (!unmergeState) return;

const stateForId = unmergeState.has(fingerprint) && unmergeState.get(fingerprint);
if (!stateForId) return;

Object.keys(stateForId).forEach(key => {
  if (stateForId[key] !== this.state[key]) return;

  this.setState({
    [key]: stateForId[key]
  });
});
```

... and I think most of the code follows a similar style. FWIW, [Martin Fowler feels similarly](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html).